### PR TITLE
Created problem_34 in project_euler

### PR DESCRIPTION
**Pull request by Kush1101**
### **Describe your change:**

* [ ] Add an algorithm?
* [ ] Fix a bug or typo in an existing algorithm?
* [ ] Documentation change?

### **Checklist:**
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [x] This pull request is all my own work -- I have not plagiarized.
* [x] I know that pull requests will not be merged if they fail the automated tests.
* [x] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [x] All new Python files are placed inside an existing directory.
* [x] All filenames are in all lowercase characters with no spaces or dashes.
* [x] All functions and variable names follow Python naming conventions.
* [x] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [x] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [ ] All new algorithms have a URL in its comments that points to Wikipedia or other similar explanation.
* [ ] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

### CODE REVIEW

**Comment by Kush1101** [comment]
How do I create the __init__.py file so that it passes the tests?
I am only failing in the __init__.py test.

File: project_euler/problem_34/sol1.py [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+def factorial(n: int) -> int:
+    """
+    Returns the factorial of n
+    >>> factorial(5)
+    120
+    >>> factorial(1)
+    1
+    >>> factorial(0)
+    1
+    """
```
Comment (cclauss) [inline project_euler/problem_34/sol1.py:20]: Please and tests for `factorial(-1)` and `factorial(1.1)` and make sure the results match `math.factorial()`.

File: project_euler/problem_34/sol1.py [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+    sum of the factorials of all digits
+    add up to the number itself.
+    >>> compute()
+    40730
+    """
+    summ = 0
+    for num in range(3, 7 * factorial(9) + 1):
+        if sum_of_digit_factorial(num) == num:
+            summ += num
+    return summ
```
Comment (cclauss) [inline project_euler/problem_34/sol1.py:54]: Please make the list and then return `sum(the_list)` as this will be easier to test.

**Comment by Kush1101** [comment]
Thank You sir for your help. Hope this gets merged 🤞.

File: project_euler/problem_34/sol1.py (hunk 1 above)
Comment (Kush1101) [reply project_euler/problem_34/sol1.py:20]: Ok will do that

Comment (Kush1101) [reply project_euler/problem_34/sol1.py:29]: I applied the changes. Please review it sir.

File: project_euler/problem_34/sol1.py [hunk 3]
```diff
@@ -41,17 +59,17 @@ def sum_of_digit_factorial(n: int) -> int:
...
     >>> compute()
     40730
     """
-    summ = 0
+    the_list = []
     for num in range(3, 7 * factorial(9) + 1):
         if sum_of_digit_factorial(num) == num:
-            summ += num
-    return summ
+            the_list.append(num)
```
Comment (cclauss) [inline project_euler/problem_34/sol1.py:71]: Could we make this list using a https://docs.python.org/3/tutorial/datastructures.html#list-comprehensions

Comment (Kush1101) [reply project_euler/problem_34/sol1.py:71]: Ok sir. Will do it

File: project_euler/problem_34/sol1.py [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+    """
+    Returns the sum of the digits in n
+    >>> sum_of_digit_factorial(15)
+    121
+    >>> sum_of_digit_factorial(0)
+    1
+    """
+    digits = list(map(int, str(n)))
+    summ = sum(factorial(digit) for digit in digits)
+    return summ
```
Comment (cclauss) [inline project_euler/problem_34/sol1.py:57]:
```suggestion
    return sum(factorial(int(digit)) for digit in str(n))
```

File: project_euler/problem_34/sol1.py [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+
+def compute() -> int:
+    """
+    Returns the sum of all numbers whose
+    sum of the factorials of all digits
+    add up to the number itself.
+    >>> compute()
+    40730
+    """
+    the_list = [
```
Comment (cclauss) [inline project_euler/problem_34/sol1.py:68]:
```suggestion
    return sum(
```

File: project_euler/problem_34/sol1.py [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+        num
+        for num in range(3, 7 * factorial(9) + 1)
+        if sum_of_digit_factorial(num) == num
+    ]
```
Comment (cclauss) [inline project_euler/problem_34/sol1.py:72]:
```suggestion
    )
```

File: project_euler/problem_34/sol1.py [hunk 7]
```diff
@@ -0,0 +1,77 @@
...
+    add up to the number itself.
+    >>> compute()
+    40730
+    """
+    return sum(
+        num
+        for num in range(3, 7 * factorial(9) + 1)
+        if sum_of_digit_factorial(num) == num
+    ]
+    return sum(the_list)
```
Comment (cclauss) [inline project_euler/problem_34/sol1.py:73]:
```suggestion
```

File: project_euler/problem_34/sol1.py [hunk 8]
```diff
@@ -0,0 +1,77 @@
...
+    """
+    Returns the sum of the digits in n
+    >>> sum_of_digit_factorial(15)
+    121
+    >>> sum_of_digit_factorial(0)
+    1
+    """
+    digits = list(map(int, str(n)))
+    summ = sum(factorial(digit) for digit in digits)
+    return sum(factorial(int(digit)) for digit in str(n))
```
Comment (cclauss) [inline project_euler/problem_34/sol1.py:57]: Delete line 55 and 56

**Comment by cclauss** [comment]
Looks good to me...  Should we land it??

**Comment by Kush1101** [comment]
If the tests pass then can you please merge it.
@cclauss 
The continuous failure is because of the tests I because of the syntax rules. I am not able to run black on my computer that is why the syntax errors are not being resolved.

**Comment by Kush1101** [comment]
@cclauss 
Can we merge now sir?

**Comment by Kush1101** [comment]
@cclauss  Thank you sir for your patience.
I know you are extremely busy an I occupied too much of your attention. 
I am a beginner sir and I will learn under mentors like you. Thanks once again sir. ✌

**Pull request merged**